pallets/parachain-staking/src/lib.rs

```diff
@@ -0,0 +1,1322 @@
+
+	#[pallet::storage]
+	#[pallet::getter(fn total)]
+	/// Total capital locked by this staking pallet
+	// TODO: Might want to use Struct instead of Tuple
+	type Total<T: Config> = StorageValue<_, (BalanceOf<T>, BalanceOf<T>), ValueQuery>;
+
+	#[pallet::storage]
+	#[pallet::getter(fn candidate_pool)]
+	/// The pool of collator candidates, each with their total backing stake
+	type CandidatePool<T: Config> = StorageValue<_, OrderedSet<Bond<T::AccountId, BalanceOf<T>>>, ValueQuery>;
+
+	#[pallet::storage]
+	#[pallet::getter(fn exit_queue)]
+	/// A queue of collators awaiting exit `BondDuration` delay after request
+	type ExitQueue<T: Config> = StorageValue<_, OrderedSet<Bond<T::AccountId, RoundIndex>>, ValueQuery>;
+
+	#[pallet::storage]
+	#[pallet::getter(fn at_stake)]
+	/// Snapshot of collator delegation stake at the start of the round
+	// TODO: Try to reduce storage footprint
+	pub type AtStake<T: Config> = StorageDoubleMap<
+		_,
+		Twox64Concat,
+		RoundIndex,
+		Twox64Concat,
+		T::AccountId,
+		CollatorSnapshot<T::AccountId, BalanceOf<T>>,
+		ValueQuery,
+	>;
+
+	#[pallet::storage]
+	#[pallet::getter(fn inflation_config)]
+	/// Inflation configuration
+	pub type InflationConfig<T: Config> = StorageValue<_, InflationInfo, ValueQuery>;
+
+	#[pallet::storage]
+	#[pallet::getter(fn reward_locks)]
+	/// Locked balance which has been granted as reward after a collator
+	/// authored a block
+	pub type RewardLocks<T: Config> =
+		StorageMap<_, Twox64Concat, T::AccountId, BTreeMap<T::BlockNumber, BalanceOf<T>>, ValueQuery>;
+
+	#[pallet::genesis_config]
+	pub struct GenesisConfig<T: Config> {
+		pub stakers: Vec<(T::AccountId, Option<T::AccountId>, BalanceOf<T>)>,
+		pub inflation_config: InflationInfo,
+	}
+
+	#[cfg(feature = "std")]
+	impl<T: Config> Default for GenesisConfig<T> {
+		fn default() -> Self {
+			Self {
+				stakers: vec![],
+				..Default::default()
+			}
+		}
+	}
+
+	#[pallet::genesis_build]
+	impl<T: Config> GenesisBuild<T> for GenesisConfig<T> {
+		fn build(&self) {
+			assert!(self.inflation_config.is_valid(), "Invalid inflation configuration");
+			<InflationConfig<T>>::put(self.inflation_config.clone());
+
+			for &(ref actor, ref opt_val, balance) in &self.stakers {
+				assert!(
+					T::Currency::free_balance(&actor) >= balance,
+					"Account does not have enough balance to bond."
+				);
+				let _ = if let Some(delegated_val) = opt_val {
+					<Pallet<T>>::delegate(
+						T::Origin::from(Some(actor.clone()).into()),
+						delegated_val.clone(),
+						balance,
+					)
+				} else {
+					<Pallet<T>>::join_candidates(T::Origin::from(Some(actor.clone()).into()), balance)
+				};
+			}
+			// Set total selected candidates to minimum config
+			<TotalSelected<T>>::put(T::MinSelectedCandidates::get());
+			// Choose top TotalSelected collator candidates
+			let (v_count, collator_staked, delegator_staked) = <Pallet<T>>::select_top_candidates(1u32);
+			// Start Round 1 at Block 0
```

Comment:
```suggestion
			// Start Round 1 at Block 0
```

a nit preference is that things should be zero based. Should be no problem to start at "round 0"